service/controller/resource/namespace/resource.go

```diff
@@ -30,6 +30,7 @@ func New(config Config) (*generic.Resource, error) {
 		ClientFunc:     clientFunc,
 		Logger:         config.Logger,
 		Name:           Name,
+		GetObjectMeta:  getObjectMeta,
 		ToCR:           toNamespace,
```

Comment:
`GetObjectMeta` and `ToCR` are quite closely related, it would be good to reflect this in their names.

I would suggest to rename `ToCR` to `GetObject`

```suggestion
		GetObjectMeta:  getObjectMeta,
		GetObject:           toNamespace,
```

Reply:
I like that. I was also thinking that it would be cool if the name reflected that this is supposed to return desired state, so e.g. `GetDesiredObject`, WDYT? If you think it's unnecessary or long I'll go with `GetObject`.

Reply:
I like `GetDesiredObject` better actually 👍